[ML] job saved objects initialization

jgowdyelastic opened the pull request:
This PR has been closed, please go to https://github.com/elastic/kibana/pull/82639

---

Adds a check during the plugin's start to see if any `ml-job` saved objects exist. If they don't, and ML jobs do exist, it will run an initialization process to create them.

Also moves the `mlLog` service to a better location.

- [x] This was checked for breaking API changes and was [labeled appropriately](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)

spalger committed:
remove entire suite as partial skips aren't doing the trick

(cherry picked from commit 4ce3e1ce03336c8b838f881a77a75d724e469315)

jgowdyelastic committed:
[ML] job saved objects initialization

elasticmachine commented:
Pinging @elastic/ml-ui (:ml)

jgowdyelastic committed:
fixing job count logic

jgowdyelastic commented on x-pack/plugins/ml/server/saved_objects/initialization.ts:
```diff
@@ -0,0 +1,96 @@
+    // const { body } = await client.asInternalUser.count({
+    //   index: '.ml-config',
+    // });
+    // return body.count > 0;
```
@droberts195 @legrego 
This function needs to check that there is at least one ML job in existence.
I believe the fastest way to do this would be to perform a count on `.ml-config`. However, the kibana system user does not have read access to that index.
Instead i'm having to call both AD and DFA get endpoints to perform the same check (below).
Do you think it would be safe to give the kibana system user the ability to make this check?

legrego replied:
I'm not terribly familiar with the contents of `.ml-config`. It appears to be a listing of jobs? Is there anything else in that index?

A more direct question might be: Is there any information in that index that's not accessible via the ML APIs granted to users with the `manage_ml` cluster privilege? I'd be ok granting `read` access to the `.ml-config` index if this is true, since `kibana_system` already has that cluster privilege.

On the other hand, if there is information in this index that the `kibana_system` user currently doesn't have access to via other APIs, then I'd need a better understanding of that data before answering one way or another.

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85554/)
* Commit: 22b04978056359b673660dbaa8050c634c757bd8
* This comment will update when the build is complete

### Failed CI Steps
* [Check Types](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85554/execution/node/375/log)

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85554/","number":"85554","commit":"22b04978056359b673660dbaa8050c634c757bd8"}]}
PIPELINE-->

jgowdyelastic committed:
adding missing files

kibanamachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85586/)
* Commit: fa95dcf1566bfff8ca3b8596773971d6929e4f63
* [Pipeline Steps](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85586/flowGraphTable) (look for red circles / failed steps)
* [Interpreting CI Failures](https://www.elastic.co/guide/en/kibana/current/interpreting-ci-failures.html)

### Failed CI Steps
* [Build X-Pack Plugins](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85586/execution/node/125/log)
* [Report failed tests, if necessary](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85586/execution/node/137/log)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
> ‼️ ERROR: metrics for fa95dcf1566bfff8ca3b8596773971d6929e4f63 were not reported

### History
* :broken_heart: [Build #85554](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85554/) failed 22b04978056359b673660dbaa8050c634c757bd8

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85554/","number":"85554","commit":"22b04978056359b673660dbaa8050c634c757bd8"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85586/","number":"85586","commit":"fa95dcf1566bfff8ca3b8596773971d6929e4f63"}]}
PIPELINE-->

jgowdyelastic commented:
@elasticmachine merge upstream

kibanamachine committed:
Merge branch 'master' into job-saved-object-initialization

scottybollinger committed:
[Workplace Search] Migrate shared LicenseBadge component (#82372)

* [Enterprise Search] Migrate shared LicenseBadge component

* Move files to Workplace Search dir

I mistakenly thought these were shared at the Enterprise Search level

kibanamachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85589/)
* Commit: 7934aa8f32ea669bf20f26ee61c90c224b5d4a7a
* [Pipeline Steps](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85589/flowGraphTable) (look for red circles / failed steps)
* [Interpreting CI Failures](https://www.elastic.co/guide/en/kibana/current/interpreting-ci-failures.html)

### Failed CI Steps
* [Build X-Pack Plugins](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85589/execution/node/125/log)
* [Report failed tests, if necessary](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85589/execution/node/137/log)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
> ‼️ ERROR: metrics for 7934aa8f32ea669bf20f26ee61c90c224b5d4a7a were not reported

### History
* :broken_heart: [Build #85586](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85586/) failed fa95dcf1566bfff8ca3b8596773971d6929e4f63
* :broken_heart: [Build #85554](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85554/) failed 22b04978056359b673660dbaa8050c634c757bd8

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85554/","number":"85554","commit":"22b04978056359b673660dbaa8050c634c757bd8"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85586/","number":"85586","commit":"fa95dcf1566bfff8ca3b8596773971d6929e4f63"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85589/","number":"85589","commit":"7934aa8f32ea669bf20f26ee61c90c224b5d4a7a"}]}
PIPELINE-->

jgowdyelastic committed:
attempting to fix build crash

jgowdyelastic committed:
Merge branch 'job-saved-object-initialization' of github.com:jgowdyelastic/kibana into job-saved-object-initialization

JasonStoltz committed:
[Enterprise Search] Added App Search log settings routes (#82162)

kibanamachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85595/)
* Commit: dc3c384a1b005f7ec23b7dba54f8e2ab052bd241
* [Pipeline Steps](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85595/flowGraphTable) (look for red circles / failed steps)
* [Interpreting CI Failures](https://www.elastic.co/guide/en/kibana/current/interpreting-ci-failures.html)

### Failed CI Steps
* [Build X-Pack Plugins](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85595/execution/node/125/log)
* [Report failed tests, if necessary](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85595/execution/node/137/log)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
> ‼️ ERROR: metrics for dc3c384a1b005f7ec23b7dba54f8e2ab052bd241 were not reported

### History
* :broken_heart: [Build #85589](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85589/) failed 7934aa8f32ea669bf20f26ee61c90c224b5d4a7a
* :broken_heart: [Build #85586](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85586/) failed fa95dcf1566bfff8ca3b8596773971d6929e4f63
* :broken_heart: [Build #85554](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85554/) failed 22b04978056359b673660dbaa8050c634c757bd8

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85554/","number":"85554","commit":"22b04978056359b673660dbaa8050c634c757bd8"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85586/","number":"85586","commit":"fa95dcf1566bfff8ca3b8596773971d6929e4f63"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85589/","number":"85589","commit":"7934aa8f32ea669bf20f26ee61c90c224b5d4a7a"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85595/","number":"85595","commit":"dc3c384a1b005f7ec23b7dba54f8e2ab052bd241"}]}
PIPELINE-->

jgowdyelastic committed:
fixing kibana.json

gchaps committed:
[DOCS] Updates index patterns docs (#81864)

* [DOCS] Updates index patterns docs

* Update docs/management/index-patterns.asciidoc

Co-authored-by: Kaarina Tungseth <kaarina.tungseth@elastic.co>

* Update docs/management/index-patterns.asciidoc

Co-authored-by: Kaarina Tungseth <kaarina.tungseth@elastic.co>

* Update docs/management/index-patterns.asciidoc

Co-authored-by: Kaarina Tungseth <kaarina.tungseth@elastic.co>

* Update docs/management/index-patterns.asciidoc

Co-authored-by: Kaarina Tungseth <kaarina.tungseth@elastic.co>

* Update docs/management/index-patterns.asciidoc

Co-authored-by: Kaarina Tungseth <kaarina.tungseth@elastic.co>

* Update docs/management/index-patterns.asciidoc

Co-authored-by: Kaarina Tungseth <kaarina.tungseth@elastic.co>

* Update docs/management/index-patterns.asciidoc

Co-authored-by: Kaarina Tungseth <kaarina.tungseth@elastic.co>

* Update docs/management/index-patterns.asciidoc

Co-authored-by: Kaarina Tungseth <kaarina.tungseth@elastic.co>

* Update docs/management/index-patterns.asciidoc

Co-authored-by: Kaarina Tungseth <kaarina.tungseth@elastic.co>

* [DOCS] Incorporates review comments

* [DOCS] Incorporated edits

Co-authored-by: Kaarina Tungseth <kaarina.tungseth@elastic.co>

Commit:
[Enterprise Search] Telemetry: refactor to Kea logic file (#81926)

* Add new TelemetryLogic helpers

- DRYs out need to import/pass http lib
- adds product-specific helpers which DRYs out an extra line

* Update all previous sendTelemetry fns to use new logic actions

* Update unit tests for updated components

- Create reusable mocks
- Update overview_logic.ts to use new Kea mock helpers (required for recent_activity.test to pass)

* Cleanup: Remove old sendTelemetry fn
+ update tests

* [PR feedback] Correctly assert the async thrown error

cjcenizal committed:
[Remote clusters] Reorganize test files (#82362)

* Refactor Remote Clusters client integration tests for readability.
- Colocate helpers with test files.
- Remove default API responses from HTTP response mocking functions to make behavior clearer at call sites.

tylersmalley committed:
skip flaky suite (#82445)

mikecote committed:
Add descriptions to alert types (#81850)

* Initial attempt at adding descriptions to alert types

* Fix typecheck failures

* Fix i18n check

* Fix failing jest test

* Fix i18n check again

* Apply changes for Uptime

* Update x-pack/plugins/apm/public/components/alerting/register_apm_alerts.ts

Co-authored-by: Casper Hübertz <casper@formgeist.com>

* Update x-pack/plugins/apm/public/components/alerting/register_apm_alerts.ts

Co-authored-by: Casper Hübertz <casper@formgeist.com>

* Fix jest test

* Update geo threshold description

* Update description of some alert types based on feedback from Gail

* Update description of some alert types based on feedback from Gail

* Fix i18n

* Fix i18n

* Fix ESLint

* Update some copy

* Update uptime alert description

* Fix typos

Co-authored-by: Casper Hübertz <casper@formgeist.com>

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85600/)
* Commit: b3079b9bf12f731be3f49f5399cc24846b8a92c4

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### async chunks size
| id | [before](https://github.com/elastic/kibana/commit/c974f56e1ca9ddd0c5f0460fd2737ae8c580716f) | [after](https://github.com/elastic/kibana/commit/b3079b9bf12f731be3f49f5399cc24846b8a92c4) | diff |
| --- | --- | --- | --- |
| `ml` | 6.6MB | 6.6MB | +462.0B |

#### distributable file count
| id | [before](https://github.com/elastic/kibana/commit/c974f56e1ca9ddd0c5f0460fd2737ae8c580716f) | [after](https://github.com/elastic/kibana/commit/b3079b9bf12f731be3f49f5399cc24846b8a92c4) | diff |
| --- | --- | --- | --- |
| `default` | 42718 | 42721 | +3 |

#### page load bundle size
| id | [before](https://github.com/elastic/kibana/commit/c974f56e1ca9ddd0c5f0460fd2737ae8c580716f) | [after](https://github.com/elastic/kibana/commit/b3079b9bf12f731be3f49f5399cc24846b8a92c4) | diff |
| --- | --- | --- | --- |
| `spaces` | 286.7KB | 286.8KB | +109.0B |

### History
* :broken_heart: [Build #85595](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85595/) failed dc3c384a1b005f7ec23b7dba54f8e2ab052bd241
* :broken_heart: [Build #85589](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85589/) failed 7934aa8f32ea669bf20f26ee61c90c224b5d4a7a
* :broken_heart: [Build #85586](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85586/) failed fa95dcf1566bfff8ca3b8596773971d6929e4f63
* :broken_heart: [Build #85554](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85554/) failed 22b04978056359b673660dbaa8050c634c757bd8

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85554/","number":"85554","commit":"22b04978056359b673660dbaa8050c634c757bd8"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85586/","number":"85586","commit":"fa95dcf1566bfff8ca3b8596773971d6929e4f63"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85589/","number":"85589","commit":"7934aa8f32ea669bf20f26ee61c90c224b5d4a7a"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85595/","number":"85595","commit":"dc3c384a1b005f7ec23b7dba54f8e2ab052bd241"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85600/","number":"85600","commit":"b3079b9bf12f731be3f49f5399cc24846b8a92c4"}]}
PIPELINE-->

Commit:
[Resolver] Screenshot the nodes of the test plugin. (#81405)

This PR adds screenshot comparison tests for the nodes in the graph on the test plugin.
Run the tests using this command:
`yarn test:ftr --config x-pack/test/plugin_functional/config.ts --grep Resolver`

patrykkopycinski committed:
[Security Solution] Fix host details query when missing IP (#78912)

Commit:
[Ingest Manager] Replace logs/metrics strings with const (#82424)

## Summary

Use new runtime & type values added in https://github.com/elastic/kibana/pull/82231 (related to #82188 ) instead of strings. Same results. Stronger type safety.

Some before/after pics below:

<img width="680" alt="Screen Shot 2020-11-03 at 6 40 04 AM" src="https://user-images.githubusercontent.com/57655/97989501-9ff2c500-1dac-11eb-8a2d-96ddadfe2477.png">
<img width="525" alt="Screen Shot 2020-11-03 at 6 40 19 AM" src="https://user-images.githubusercontent.com/57655/97989503-a08b5b80-1dac-11eb-97ee-371a0487fbd3.png">

<img width="790" alt="Screen Shot 2020-11-03 at 7 47 42 AM" src="https://user-images.githubusercontent.com/57655/97989505-a08b5b80-1dac-11eb-9db9-54a2719a85b2.png">
<img width="738" alt="Screen Shot 2020-11-03 at 7 47 34 AM" src="https://user-images.githubusercontent.com/57655/97989504-a08b5b80-1dac-11eb-8d1d-692feafd5fe3.png">


<img width="778" alt="Screen Shot 2020-11-03 at 7 54 17 AM" src="https://user-images.githubusercontent.com/57655/97989507-a123f200-1dac-11eb-879d-ab9be4e060eb.png">
<img width="813" alt="Screen Shot 2020-11-03 at 7 54 45 AM" src="https://user-images.githubusercontent.com/57655/97989508-a123f200-1dac-11eb-8725-c44068d6b233.png">

mbondyra committed:
[Lens] Drag within dimension group to reorder (#80547)

scottybollinger committed:
[Enterprise Search] Migrate shared CredentialItem component (#82471)

tylersmalley committed:
Prevent incompatable upgrade of crypto-policies (#82475)

An upgrade of crypto-policies is incompatible with libnss as it removes
`/usr/bin/update-crypto-policies` being used.

A follow-up issue has been created to track removing this pinning https://github.com/elastic/kibana/issues/82474

Signed-off-by: Tiago Costa <tiago.costa@elastic.co>
Signed-off-by: Tyler Smalley <tyler.smalley@elastic.co>

tylersmalley committed:
skip flaky suite (#75127)

cjcenizal committed:
Add more prompts to 8.0 breaking change issue template. (#82184)

Commit:
[ci] run server integration tests (#81698)

Co-authored-by: spalger <spalger@users.noreply.github.com>
Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

Commit:
Switch to new 'as const' + typeof + ValueOf approach (#82499)

## Summary

Follow pattern from #82188. Same behavior as now. Stronger type safety.

 * `installationSources` is runtime value (JS object)
 * `InstallationSource` is TS type (`typeof installationSources`)
 * `ValueOf<InstallationSource>` is TS union type `'installed' | 'not_installed'`
 * Can access values directly as `installationSources.Installed` (JS) or `InstallationSource['Installed']` (TS)

FrankHassanabad committed:
Fixes ui strings around threat matching/mapping (#82510)

## Summary

Fixes the UI strings around the threat matching feature.

* Changes the wording in area of `threat query` to `Indicator index query`
* Uses sentence casing according to the [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing) instead of capitals on each word which was inconsistent.

Before in the about box:
<img width="1188" alt="Screen Shot 2020-11-03 at 1 57 06 PM" src="https://user-images.githubusercontent.com/1151048/98040659-dd237b00-1ddd-11eb-9750-7cf316deddd0.png">

After:
<img width="1212" alt="Screen Shot 2020-11-03 at 2 02 31 PM" src="https://user-images.githubusercontent.com/1151048/98040684-e6144c80-1ddd-11eb-90b4-853b97678618.png">

Before when creating/editing a threat match where I was using capital letters that were inconsistent:
<img width="1165" alt="Screen Shot 2020-11-03 at 2 08 33 PM" src="https://user-images.githubusercontent.com/1151048/98040774-18be4500-1dde-11eb-8c83-b7562191644b.png">

After where I use sentence case:
<img width="1054" alt="Screen Shot 2020-11-03 at 2 02 20 PM" src="https://user-images.githubusercontent.com/1151048/98040820-2c69ab80-1dde-11eb-96fb-c04644e90e3a.png">


### Checklist

- [x] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/master/packages/kbn-i18n/README.md)

brokensound77 committed:
[Detection Rules] Add 7.10 rules - v3 (#82214)

spong committed:
[Security Solution][Detections] Fixes bulk alert status update to only update alerts within the defined daterange (#82071)

## Summary

Addresses https://github.com/elastic/kibana/issues/82004 where updating an alerts status would result in all alerts being updated as the request is sent without the necessary daterange filter.

##### Before:
<details><summary>Query</summary>
<p>

``` json
  {
    "conflicts": "proceed",
    "status": "open",
    "query": {
      "bool": {
        "must": [],
        "filter": [
          { "match_all": {} },
          { "term": { "signal.status": "closed" } }
        ],
        "should": [],
        "must_not": [
          { "exists": { "field": "signal.rule.building_block_type" } }
        ]
      }
    }
  }
```
</p>
</details>
<p align="center">
  <img width="500" src="https://user-images.githubusercontent.com/2946766/97628470-5db73580-19f2-11eb-8e51-61e428e7804f.gif" />
</p>


##### After:
<details><summary>Query</summary>
<p>

``` json
  {
    "conflicts": "proceed",
    "status": "closed",
    "query": {
      "bool": {
        "must": [],
        "filter": [
          {
            "bool": {
              "filter": [
                {
                  "bool": {
                    "should": [
                      {
                        "range": {
                          "@timestamp": { "gte": "2020-10-29T20:17:23.357Z" }
                        }
                      }
                    ],
                    "minimum_should_match": 1
                  }
                },
                {
                  "bool": {
                    "should": [
                      {
                        "range": {
                          "@timestamp": { "lte": "2020-10-29T20:17:40.097Z" }
                        }
                      }
                    ],
                    "minimum_should_match": 1
                  }
                }
              ]
            }
          },
          { "term": { "signal.status": "open" } }
        ],
        "should": [],
        "must_not": [
          { "exists": { "field": "signal.rule.building_block_type" } }
        ]
      }
    }
  }
```
</p>
</details>
<p align="center">
  <img width="500" src="https://user-images.githubusercontent.com/2946766/97628955-0796c200-19f3-11eb-9ec3-2a6ace17160b.gif" />
</p>



### Checklist

Delete any items that are not applicable to this PR.

- [x] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios

qn895 committed:
[ML] Add probability values in decision path visualization for classification data frame analytics (#80229)

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

tylersmalley committed:
[test] Updates rollup test to allow incoming doc_count field mapper (#82547)

* [test] Updates rollup test to allow incoming doc_count field mapper

A doc_count field mapper was added in
https://github.com/elastic/elasticsearch/pull/64503

This is currently failing the ES promotion. After this promotion, we can
go back and update this test to be an exact match on the body if we
think that is the desired assertion.

Signed-off-by: Tyler Smalley <tyler.smalley@elastic.co>

jportner committed:
Fix saved object share UI bugs regarding read-only privileges (#81828)

mshustov committed:
clarify what "deps" term (#82465)

Dosant committed:
[Search][Dashboard] Restore searchSessionId from URL (#81489)

flash1293 committed:
[Lens] Add functional test for wrapping labels (#82453)

* add functional test for wrapping labels

* fix unused import

darnautov commented on x-pack/plugins/ml/server/saved_objects/initialization.ts:
```diff
@@ -0,0 +1,96 @@
+        const savedObjectsClient = getInternalSavedObjectsClient(core);
+        const jobSavedObjectService = jobSavedObjectServiceFactory(savedObjectsClient);
+        const { initSavedObjects } = repairFactory(
+          (client as unknown) as IScopedClusterClient,
```
maybe move this typecasting to the `client` definition itself?

darnautov commented on x-pack/plugins/ml/server/saved_objects/repair.ts:
```diff
@@ -0,0 +1,217 @@
+
+    const status = await checkStatus();
+    for (const job of status.jobs['anomaly-detector']) {
+      if (job.checks.savedObjectExits === false) {
+        if (simulate === true) {
+          results.savedObjectsCreated[job.jobId] = { success: true };
+        } else {
+          // create AD saved objects for jobs which are missing them
+          const jobId = job.jobId;
+          const datafeedId = job.datafeedId;
+          tasks.push(async () => {
+            try {
+              await jobSavedObjectService.createAnomalyDetectionJob(jobId, datafeedId ?? undefined);
+              results.savedObjectsCreated[job.jobId] = { success: true };
+            } catch (error) {
+              results.savedObjectsCreated[job.jobId] = {
+                success: false,
+                error: error.body ?? error,
+              };
+            }
+          });
+        }
+      }
```
nit: Avoid extra indentation and make readability a bit better. Applicable for the `for...of` statement below as well 
```suggestion
if (job.checks.savedObjectExits) return;

if (simulate === true) {
    results.savedObjectsCreated[job.jobId] = {success: true};
    return;
}

// create AD saved objects for jobs which are missing them
const jobId = job.jobId;
const datafeedId = job.datafeedId;
tasks.push(async () => {
    try {
        await jobSavedObjectService.createAnomalyDetectionJob(jobId, datafeedId ?? undefined);
        results.savedObjectsCreated[job.jobId] = {success: true};
    } catch (error) {
        results.savedObjectsCreated[job.jobId] = {
            success: false,
            error: error.body ?? error,
        };
    }
});
```

jgowdyelastic replied:
`return` breaks the loop, these would need to be `continue`

jgowdyelastic replied:
i agree that these changes would improve the readability of this code, however this function is not new and until we have tests to cover it i'd like to avoid refactoring that could introduce bugs.

darnautov commented on x-pack/plugins/ml/server/saved_objects/repair.ts:
```diff
@@ -0,0 +1,217 @@
+) {
+  const { checkStatus } = checksFactory(client, jobSavedObjectService);
+
+  async function repairJobs(simulate: boolean = false) {
```
when `simulate` is `true`, shall we just return a default object with success flags?

jgowdyelastic replied:
see comment below about future refactoring

cnasikas committed:
[Security Solution][Case] Case action type (#80870)

* Init connector

* Add test

* Improve comment type

* Add integration tests

* Fix i18n

* Improve tests

* Show unknown when username is null

* Improve comment type

* Pass connector to case client

* Improve type after PR #82125

* Add comment migration test

* Fix integration tests

* Fix reporter on table

* Create case connector ui

* Add connector to README

* Improve casting on executor

* Translate name

* Improve test

* Create comment type enum

* Fix type

* Fix i18n

* Move README to cases

* Filter out case connector from alerting

Co-authored-by: Mike Côté <mikecote@users.noreply.github.com>

Co-authored-by: Mike Côté <mikecote@users.noreply.github.com>

flash1293 committed:
[Lens] Categorical color palettes (#75309)

dmlemeshko committed:
[test/new_visualize_flow] fix config (#82441)

* [test/new_visualize_flow] fix config

* fix file extension

mbondyra committed:
Add moving average function  (#82122)

scottybollinger committed:
[Enterprise Search] Migrate shared ApiKey component (#82511)

scottybollinger committed:
[Enterprise Search] Migrate shared ComponentLoader component (#82504)

peteharverson commented on x-pack/plugins/ml/server/saved_objects/initialization.ts:
```diff
@@ -0,0 +1,96 @@
+            filter: [
+              {
+                term: {
+                  type: 'ml-job',
```
Use `ML_SAVED_OBJECT_TYPE ` from ` x-pack/plugins/ml/common/types/saved_objects.ts ` here?

peteharverson commented on x-pack/plugins/ml/server/saved_objects/repair.ts:
```diff
@@ -0,0 +1,217 @@
+        }
+      }
+    }
+    for (const job of status.jobs['data-frame-analytics']) {
```
Do we have consts / enums for `data-frame-analytics` and `anomaly-detector`?

jgowdyelastic replied:
I chose to go with making `JobType` a string union as it's far easier to work with.
Typescript ensures only `data-frame-analytics` and `anomaly-detector` can be used.

peteharverson reviewed (approved):
Tested and LGTM. Just left a couple of minor comments.

Commit:
[Ingest Manager] Switch DefaultPackage enum to approach from #82188 (#82512)

## Summary

Follow pattern from #82188. Same behavior as now. Stronger type safety.

 * `defaultPackages` is runtime value (JS object)
 * `DefaultPackage` is TS type (`typeof defaultPackages`)
 * `ValueOf<DefaultPackage>` is TS union type `'system' | 'endpoint'`
 * Can access values directly as `defaultPackages.System` (JS) or `DefaultPackage['System']` (TS)

pgayvallet committed:
SavedObjects management: display explicit import error in case of failure and avoid invalid file to crash the server (#82406)

* display error cause in case of import failure

* avoid server crash when importing invalid file

* remove unused translations

* fix unit tests

* change savedObjects.maxImportPayloadBytes default to 25mb

* fix types and logic

mattkime committed:
Index patterns api - load field list on server  (#81218)

Index patterns api - load field list on server

brianseeders committed:
[CI] Move notifyOnError to fix in-progress alerts for ciGroups (#82506)

ryankeairns committed:
Change monaco background color to match inputs (#82451)

Commit:
[Enterprise Search] Trivial API fixes (#82552)

* Fix embarrassing typo

* Fix uncovered line

smith committed:
TS project references for securityOss plugin (#82135)

jgowdyelastic committed:
Merge branches 'job-saved-object-initialization' and 'job-saved-object-initialization' of github.com:jgowdyelastic/kibana into job-saved-object-initialization

jgowdyelastic committed:
changes based on review

elasticmachine commented:
Pinging @elastic/ingest-management (Team:Ingest Management)

elasticmachine commented:
Pinging @elastic/apm-ui (Team:apm)

jgowdyelastic commented:
A slip of the finger completely broke this PR.
Closing and creating a new one.

jgowdyelastic commented on x-pack/plugins/ml/server/saved_objects/initialization.ts:
```diff
@@ -0,0 +1,96 @@
+import { jobSavedObjectServiceFactory } from './service';
+import { mlLog } from '../lib/log';
+
+export function jobInitializationFactory(core: CoreStart) {
```
changed in https://github.com/elastic/kibana/pull/82450/commits/2dae54e37041ff1cb6c92db9b1925c0c18ae377c

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85884/)
* Commit: 2dae54e37041ff1cb6c92db9b1925c0c18ae377c
* This comment will update when the build is complete

### Failed CI Steps
* [Execute oss-pluginFunctional](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85884/execution/node/511/log)
* [Execute oss-pluginFunctional](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85884/execution/node/703/log)

### History
* :green_heart: [Build #85600](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85600/) succeeded b3079b9bf12f731be3f49f5399cc24846b8a92c4
* :broken_heart: [Build #85595](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85595/) failed dc3c384a1b005f7ec23b7dba54f8e2ab052bd241
* :broken_heart: [Build #85589](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85589/) failed 7934aa8f32ea669bf20f26ee61c90c224b5d4a7a
* :broken_heart: [Build #85586](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85586/) failed fa95dcf1566bfff8ca3b8596773971d6929e4f63
* :broken_heart: [Build #85554](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85554/) failed 22b04978056359b673660dbaa8050c634c757bd8

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85554/","number":"85554","commit":"22b04978056359b673660dbaa8050c634c757bd8"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85586/","number":"85586","commit":"fa95dcf1566bfff8ca3b8596773971d6929e4f63"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85589/","number":"85589","commit":"7934aa8f32ea669bf20f26ee61c90c224b5d4a7a"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85595/","number":"85595","commit":"dc3c384a1b005f7ec23b7dba54f8e2ab052bd241"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85600/","number":"85600","commit":"b3079b9bf12f731be3f49f5399cc24846b8a92c4"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/85884/","number":"85884","commit":"2dae54e37041ff1cb6c92db9b1925c0c18ae377c"}]}
PIPELINE-->
